src/rebuild/srv.c

```diff
@@ -1131,30 +1131,24 @@ rebuild_try_merge_tgts(struct ds_pool *pool, uint32_t map_ver,
 	 * complete.
 	 */
 	d_list_for_each_entry(task, &rebuild_gst.rg_queue_list, dst_list) {
-		if (uuid_compare(task->dst_pool_uuid, pool->sp_uuid) != 0)
-			/* This task isn't for this pool - don't consider it */
-			continue;
+		if (uuid_compare(task->dst_pool_uuid, pool->sp_uuid) != 0) {
+			if (merge_task == NULL)
+				continue;
+			break;
+		}
```

Comment:
I think we need to check rebuild_op inside the loop, otherwise, we may miss the merge chance. For example:

[T1, pool_1, v1, reint] [T2, pool_2, v2, rebuild] [T3, pool_1, v3, rebuild], now T4 coming in for pool_1 with rebuild.

It is expected the T3 and T4 can merge. But based on current logic, it merge nothing.

Reply:
Hmm, all job for the same pool should be together,  so T3 should be ahead of T2, and together with T1.